### ISSUE TITLE
DecomposeBoxes doesn't decompose CircBox with nested classical control

### ISSUE THREAD

**Issue by sjdilkes** [CQCL/tket#1583]
See: 
```
from pytket.circuit import CircBox, Circuit, Bit, Qubit
from pytket.qasm import circuit_to_qasm_str
from pytket.passes import DecomposeBoxes

c_inner: Circuit = Circuit(4, 1)
cbox: CircBox = CircBox(Circuit(4).H(0).H(1).H(2).H(3).CX(0, 1).CX(1, 2).CX(2, 3))
c_inner.add_circbox(cbox, [Qubit(i) for i in range(4)], condition=Bit(0))

c_outer: Circuit = Circuit(4, 2)
c_outer.add_circbox(CircBox(c_inner), c_inner.qubits + c_inner.bits, condition=Bit(1))

DecomposeBoxes().apply(c_outer)
print(circuit_to_qasm_str(c_outer, header="hqslib1"))
```

This code gives an error as `c_outer` still has a `CircBox` internally. Printing the circuit commands will give `IF ([c[1]] == 1) THEN IF ([c[0]] == 1) THEN CircBox q[0], q[1], q[2], q[3];`.

**Comment by sjdilkes** [comment]
Alternatively, applying `auto_rebase_pass({OpType.ZZPhase, OpType.PhasedX, OpType.Rz}).apply(c_outer)` to the circuit throws a `BadOpType` error, complaining about a conditional operation.

**Comment by cqc-alec** [comment]
I have a fix for the `DecomposeBoxes` issue (PR coming soon), but even after the fix it doesn't seem possible to convert to QASM. I've created a separate issue for this ( https://github.com/CQCL/tket/issues/1745 ).

**Comment by sjdilkes** [comment]
> I have a fix for the `DecomposeBoxes` issue (PR coming soon), but even after the fix it doesn't seem possible to convert to QASM. I've created a separate issue for this ( [#1745](https://github.com/CQCL/tket/issues/1745) ).
> 
> EDIT: Turns out nested conditionals are just not supported by QASM at all.

If they aren't supported then it sounds like we can close this issue once https://github.com/CQCL/tket/pull/1746 is merged?

**Comment by cqc-alec** [comment]
> If they aren't supported then it sounds like we can close this issue once [#1746](https://github.com/CQCL/tket/pull/1746) is merged?

Yes, though I may open another one for the rebase failure.

**Comment by cqc-alec** [comment]
> > If they aren't supported then it sounds like we can close this issue once [#1746](https://github.com/CQCL/tket/pull/1746) is merged?
> 
> Yes, though I may open another one for the rebase failure.

Ah, there is already #1727 .
